### DISCUSSION TITLE
How to fix "ERROR:type SBM has no field soil"?

### DISCUSSION THREAD

**Discussion by DavidChoi76** [Deltares/Wflow.jl#494] [Q&A] [answered]
I have started studying Wflow with [Wflow.jl document](https://deltares.github.io/Wflow.jl/dev/user_guide/step4_running/) 
When I executed "sbm_config.toml" below, I got an error like "ERROR: type SBM has no field soil". Could you help me to solve this problem?

'''
$ **julia -e "using Wflow; Wflow.run()" sbm_config.toml**
[ Info: Wflow version v0.8.1
[ Info: Initialize model variables for model type sbm.
[ Info: Cyclic parameters are provided by data\input\staticmaps-moselle.nc.
[ Info: Forcing parameters are provided by data\input\forcing-moselle.nc.
[ Info: Set vertical.atmospheric_forcing.precipitation using netCDF variable precip as forcing parameter.
[ Info: Set vertical.atmospheric_forcing.temperature using netCDF variable temp as forcing parameter.
[ Info: Set vertical.atmospheric_forcing.potential_evaporation using netCDF variable pet as forcing parameter.
[ Info: Set vertical.vegetation_parameter_set.leaf_area_index using netCDF variable LAI as cyclic parameter, with 12 timesteps.
┌ Info: General model settings
│   reservoirs = true
│   lakes = false
│   snow = true
│   masswasting = true
└   glacier = false
[ Info: Set subcatchment using netCDF variable wflow_subcatch.
[ Info: Set river_location using netCDF variable wflow_river.
[ Info: Set lateral.river.width using netCDF variable wflow_riverwidth.
[ Info: Set lateral.river.length using netCDF variable wflow_riverlength.
[ Info: Set vertical.cfmax using default value 3.75653.
[ Info: Set vertical.tt using default value 0.0.
[ Info: Set vertical.tti using default value 1.0.
[ Info: Set vertical.ttm using default value 0.0.
[ Info: Set vertical.whc using default value 0.1.
[ Info: Set vertical.w_soil using default value 0.1125.
[ Info: Set vertical.cf_soil using default value 0.038.
[ Info: Set vertical.g_tt using default value 0.0.
[ Info: Set vertical.g_cfmax using default value 3.0.
[ Info: Set vertical.g_sifrac using default value 0.001.
[ Info: Set vertical.glacierfrac using default value 0.0.
[ Info: Set vertical.glacierstore using default value 5500.0.
[ Info: Set vertical.theta_s using default value 0.6.
[ Info: Set vertical.theta_r using default value 0.01.
[ Info: Set vertical.kv_0 using default value 3000.0.
[ Info: Set vertical.f using default value 0.001.
[ Info: Set vertical.hb using default value -10.0.
[ Info: Set vertical.h1 using default value 0.0.
[ Info: Set vertical.h2 using default value -100.0.
[ Info: Set vertical.h3_high using default value -400.0.
[ Info: Set vertical.h3_low using default value -1000.0.
[ Info: Set vertical.h4 using default value -15849.0.
[ Info: Set vertical.alpha_h1 using default value 1.0.
[ Info: Set vertical.soilthickness using default value 2000.0.
[ Info: Set vertical.infiltcappath using default value 10.0.
[ Info: Set vertical.infiltcapsoil using default value 100.0.
[ Info: Set vertical.maxleakage using default value 0.0.
[ Info: Set vertical.c using default value 10.0.
[ Info: Set vertical.kvfrac using default value 1.0.
[ Info: Set vertical.waterfrac using default value 0.0.
[ Info: Set vertical.pathfrac using default value 0.01.
[ Info: Set vertical.rootingdepth using default value 750.0.
[ Info: Set vertical.rootdistpar using default value -500.0.
[ Info: Set vertical.cap_hmax using default value 2000.0.
[ Info: Set vertical.cap_n using default value 2.0.
[ Info: Set vertical.kc using default value 1.0.
[ Info: Set vertical.cmax using default value 1.0.
[ Info: Set vertical.eoverr using default value 0.1.
[ Info: Set vertical.canopygapfraction using default value 0.1.
[ Info: Set lateral.river.reservoir.locs using netCDF variable wflow_reservoirlocs.
[ Info: Set lateral.river.reservoir.areas using netCDF variable wflow_reservoirareas.
[ Info: Set lateral.river.reservoir.demand using netCDF variable ResDemand.
[ Info: Set lateral.river.reservoir.maxrelease using netCDF variable ResMaxRelease.
[ Info: Set lateral.river.reservoir.maxvolume using netCDF variable ResMaxVolume.
[ Info: Set lateral.river.reservoir.area using netCDF variable ResSimpleArea.
[ Info: Set lateral.river.reservoir.targetfullfrac using netCDF variable ResTargetFullFrac.
[ Info: Set lateral.river.reservoir.targetminfrac using netCDF variable ResTargetMinFrac.
[ Info: Read 2 reservoir locations.
[ Info: Set ldd using netCDF variable wflow_ldd.
[ Info: Set lateral.land.slope using netCDF variable Slope.
[ Info: Set lateral.subsurface.ksathorfrac using netCDF variable KsatHorFrac.
┌ Info: Kinematic wave approach is used for overland flow.
└   iterate = true
[ Info: Set lateral.land.n using netCDF variable N.
┌ Info: Kinematic wave approach is used for river flow.
└   iterate = true
[ Info: Set lateral.river.n using netCDF variable N_River.
[ Info: Set lateral.river.bankfull_depth using netCDF variable RiverDepth.
[ Info: Set lateral.river.slope using netCDF variable RiverSlope.
[ Info: Create an output netCDF file data\output\output_moselle.nc for grid data, using compression level 0.
**ERROR: type SBM has no field soil**
Stacktrace:
  [1] getproperty
    @ .\Base.jl:37 [inlined]
  [2] BottomRF
    @ .\reduce.jl:86 [inlined]
  [3] afoldl(::Base.BottomRF{typeof(getproperty)}, ::@NamedTuple{vertical::Wflow.SBM{Float64, 4, 5, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing}, lateral::@NamedTuple{subsurface::Wflow.LateralSSF{Float64}, land::Wflow.SurfaceFlowLand{Float64}, river::Wflow.SurfaceFlowRiver{Float64, Wflow.SimpleReservoir{Float64}, Nothing, Nothing}}}, ::Symbol, ::Symbol, ::Symbol, ::Symbol)
    @ Base .\operators.jl:545
  [4] _foldl_impl
    @ .\reduce.jl:68 [inlined]
  [5] foldl_impl
    @ .\reduce.jl:48 [inlined]
  [6] mapfoldl_impl
    @ .\reduce.jl:44 [inlined]
  [7] mapfoldl
    @ .\reduce.jl:175 [inlined]
  [8] foldl
    @ .\reduce.jl:198 [inlined]
  [9] param(obj::@NamedTuple{vertical::Wflow.SBM{Float64, 4, 5, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing}, lateral::@NamedTuple{subsurface::Wflow.LateralSSF{Float64}, land::Wflow.SurfaceFlowLand{Float64}, river::Wflow.SurfaceFlowRiver{Float64, Wflow.SimpleReservoir{Float64}, Nothing, Nothing}}}, fields::NTuple{4, Symbol})
    @ Wflow C:\Users\kwater\.julia\packages\Wflow\UKSXZ\src\io.jl:17
 [10] out_map(ncnames_dict::Dict{Tuple{Symbol, Vararg{Symbol}}, String}, modelmap::@NamedTuple{vertical::Wflow.SBM{Float64, 4, 5, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing}, lateral::@NamedTuple{subsurface::Wflow.LateralSSF{Float64}, land::Wflow.SurfaceFlowLand{Float64}, river::Wflow.SurfaceFlowRiver{Float64, Wflow.SimpleReservoir{Float64}, Nothing, Nothing}}})
    @ Wflow C:\Users\kwater\.julia\packages\Wflow\UKSXZ\src\io.jl:892
 [11] prepare_writer(config::Wflow.Config, modelmap::@NamedTuple{vertical::Wflow.SBM{Float64, 4, 5, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing}, lateral::@NamedTuple{subsurface::Wflow.LateralSSF{Float64}, land::Wflow.SurfaceFlowLand{Float64}, river::Wflow.SurfaceFlowRiver{Float64, Wflow.SimpleReservoir{Float64}, Nothing, Nothing}}}, rev_inds::@NamedTuple{land::Matrix{Int64}, river::Matrix{Int64}, reservoir::Matrix{Int64}, lake::Nothing}, x_nc::Vector{Float64}, y_nc::Vector{Float64}, nc_static::NCDatasets.NCDataset{Nothing, Missing}; extra_dim::@NamedTuple{name::String, value::Vector{Float64}})
    @ Wflow C:\Users\kwater\.julia\packages\Wflow\UKSXZ\src\io.jl:937
 [12] initialize_sbm_model(config::Wflow.Config)
    @ Wflow C:\Users\kwater\.julia\packages\Wflow\UKSXZ\src\sbm_model.jl:322
 [13] run(config::Wflow.Config)
    @ Wflow C:\Users\kwater\.julia\packages\Wflow\UKSXZ\src\Wflow.jl:189
 [14] (::Wflow.var"#284#285"{IOStream, Bool, Wflow.Config})()
    @ Wflow C:\Users\kwater\.julia\packages\Wflow\UKSXZ\src\Wflow.jl:168
 [15] with_logstate(f::Function, logstate::Any)
    @ Base.CoreLogging .\logging.jl:515
 [16] with_logger
    @ .\logging.jl:627 [inlined]
 [17] run(tomlpath::String; silent::Nothing)
    @ Wflow C:\Users\kwater\.julia\packages\Wflow\UKSXZ\src\Wflow.jl:164
 [18] run
    @ C:\Users\kwater\.julia\packages\Wflow\UKSXZ\src\Wflow.jl:156 [inlined]
 [19] run()
    @ Wflow C:\Users\kwater\.julia\packages\Wflow\UKSXZ\src\Wflow.jl:276
 [20] top-level scope
    @ none:1
'''

**Comment by JoostBuitink** [comment, marked as answer]
This is probably because we started with updating the master branch with our v1.0 changes. I am guessing that you downloaded the toml file using the code provided on this page, right? https://deltares.github.io/Wflow.jl/dev/user_guide/sample_data/#wflow_sbm_data

If so, it is probably wise to change the `toml_url` to `toml_url = "https://raw.githubusercontent.com/Deltares/Wflow.jl/v0.8.1/test/sbm_config.toml"`. Otherwise, you can browse the repository at the state of the last release (v0.8.1) via this link: https://github.com/Deltares/Wflow.jl/tree/v0.8.1

This should solve your issue!

**Reply by DavidChoi76** [reply]
That works for me. Thanks.
